include/hip/hcc_detail/program_state.hpp

```diff
@@ -386,7 +386,7 @@ const std::unordered_map<
     static std::once_flag f;
 
     std::call_once(f, []() {
-        for (auto&& agent : hip_impl::all_hsa_agents()) {
+            auto agent = current_hsa_agent();
```

Comment:
@jeffdaily I don't think this is correct. This lambda would only be invoked once per shared lib loaded. Using `current_hsa_agent` here would make the executables loaded only on 1 HSA agent, ignoring all other available agents, or those specified with `HIP_VISIBLE_DEVICES`.

Would it be a good idea to change `all_hsa_agents()` to `visible_hsa_agents()`, and leverage the implementation of `visible_accelerators`?